internal/benchrunner/runners/rally/runner.go

```diff
@@ -274,6 +341,47 @@ func (r *runner) setUp() error {
 	return nil
 }
 
+func (r *runner) extractSimulatedTemplate(indexTemplate string) (string, error) {
+	simulateTemplate, err := r.options.ESAPI.Indices.SimulateTemplate(r.options.ESAPI.Indices.SimulateTemplate.WithName(indexTemplate))
```

Comment:
Now that I played around with this API (https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-simulate-template.html) I have a much better understanding what it does. Basically it means we get the full template / current state. As you describe further down, we loose the support to use `@custom`. 

My long term preference is still having all the assets but this gets us to what we need short term. To make the @custom work, we could either do it with all the asset export we discussed yesterday or that we manually add it to the template again.